test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,479 @@
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Bad: Instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(150, SpvOpAtomicLoad, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: Operand index does not equal 2 or 3.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 1, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The Instruction is not an atomic operation.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(14, SpvOpAccessChain, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The SpvOpAtomicLoad takes one memory semantic value and the operand
+  // index passed is 1.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 1, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The new Id instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 101)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The new Id instruction exists, but not OpConstant.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 13)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The OpAtomicStore memory semantics value cannot be AcquireRelease.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(24, SpvOpAtomicStore, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The OpAtomicExchange(read-modify-write) memory semantics value cannot
+  // be changed from Acquire to Release.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicExchange, 0), 2, 99)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The SpvOpAtomicCompareExchange(read-modify-write) takes two memory
+  // semantics, one of the values must be stronger than the other.
+  ASSERT_FALSE(
+      TransformationChangingMemorySemantics(
+          MakeInstructionDescriptor(25, SpvOpAtomicCompareExchange, 0), 2, 100)
+          .IsApplicable(context.get(), transformation_context));
```

Comment:
The comment (and probably the assert) is not quite right, as I have highlighted in the transformation code. You will probably want to write a few additional tests here, once you have addressed my above comments about this.

Reply:
Improved.